### PR TITLE
Fix SVG Upload, do not check dimensions if file is svg.

### LINKED ISSUE #29292: BO - A warnings are displayed in the file manager when add an SVG image extension with php8.1
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

A warnings are displayed in the file manager when add an SVG image extension with php8.1

**8.0.x with php8.1 : NOK** :x: 

image

https://user-images.githubusercontent.com/92912932/183698943-fb20f6ad-e3bb-48ea-b1eb-f78b48239ccd.mp4

**8.0.x with php7.3 : OK** :heavy_check_mark: 

https://user-images.githubusercontent.com/92912932/183698966-09de47ff-971f-450c-a1db-c806ca3334e9.mp4

### Expected behavior

No warning is displayed

### Steps to reproduce

1. Check with php8.1
2. Go to BO > Catalog > Categories page OR Products page
3. Edit any Category/Product
4. Insert an image with SVG extension in the file manager
5. Back to the list of displayed images in the file manager
6. See error >> A warnings are displayed

### PrestaShop version(s) where the bug happened

8.0.x

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  8.0.x 
| Description?      | PHP 8.1 displays warning when you want to get dimensions on an svg file. This PR prevents it.
| Type?             | bug fix
| Category?         |  BO
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29292

### CODE REVIEW

File: admin-dev/filemanager/include/utils.php [hunk 1]
```diff
@@ -80,16 +97,22 @@ function create_img_gd($imgfile, $imgthumb, $newwidth, $newheight="")
...
 function create_img($imgfile, $imgthumb, $newwidth, $newheight="")
 {
+    if (getFileExtension($imgfile) === 'svg') {
+        copy($imgfile, $imgthumb);
+
+        return true;
+    }
+
     if (image_check_memory_usage($imgfile, $newwidth, $newheight)) {
         require_once 'php_image_magician.php';
```
Comment (matks) [inline admin-dev/filemanager/include/utils.php:113]: Outch that is a little oldie 😅

File: admin-dev/filemanager/include/utils.php [hunk 2]
```diff
@@ -64,8 +64,25 @@ function rename_folder($old_path, $name, $transliteration)
     }
 }
 
+function getFileExtension($path)
```
Comment (matks) [inline admin-dev/filemanager/include/utils.php:75]: Little phpDoc to help developer use it ? 😄

**Review by matks — DISMISSED** [review]
Good PR

I added a nice-to-have suggestion

**Review by kpodemski — APPROVED** [review]

**Review by kpodemski — DISMISSED** [review]
@FabienPapet could you update the PR table?

**Comment by FabienPapet** [comment]
@kpodemski done

File: admin-dev/filemanager/include/utils.php (hunk 2 above)
Comment (FabienPapet) [reply admin-dev/filemanager/include/utils.php:75]: I put it as internal and added some documentation.

**Comment by kpodemski** [comment]
@FabienPapet, whoops! one more thing, if you want to target `8.0.x` you should change the branch

btw. maybe you can copy "how to test" from the issue, too 👍

**Comment by matks** [comment]
@FabienPapet PHP CS Fixer is complaining 

```
---------- begin diff ----------
--- /home/runner/work/PrestaShop/PrestaShop/tests/Integration/Behaviour/Features/Context/Domain/StateFeatureContext.php
+++ /home/runner/work/PrestaShop/PrestaShop/tests/Integration/Behaviour/Features/Context/Domain/StateFeatureContext.php
@@ -87,7 +87,7 @@
             ));
 
             SharedStorage::getStorage()->set($stateReference, $stateId->getValue());
-        } catch (CannotAddStateException | StateConstraintException | StateException $e) {
+        } catch (CannotAddStateException|StateConstraintException|StateException $e) {
             $this->setLastException($e);
         }
     }
@@ -[12](https://github.com/PrestaShop/PrestaShop/runs/8020933348?check_suite_focus=true#step:9:13)3,7 +123,7 @@
 
         try {
             $this->getCommandBus()->handle($command);
-        } catch (CannotUpdateStateException | StateConstraintException | StateException | StateNotFoundException $e) {
+        } catch (CannotUpdateStateException|StateConstraintException|StateException|StateNotFoundException $e) {
             $this->setLastException($e);
         }
     }

      ----------- end diff -----------


Checked all files in 178.[16](https://github.com/PrestaShop/PrestaShop/runs/8020933348?check_suite_focus=true#step:9:17)8 seconds, 52.000 MB memory used
Error: Process completed with exit code 8.
```

File: admin-dev/filemanager/include/utils.php [hunk 3]
```diff
@@ -64,8 +64,25 @@ function rename_folder($old_path, $name, $transliteration)
...
+{
+    if (!is_file($path)) {
+        return null;
+    }
+
+    $infos = pathinfo($path);
+
+    return array_key_exists('extension', $infos) ? $infos['extension']: null;
```
Comment (atomiix) [inline admin-dev/filemanager/include/utils.php]:
```suggestion
    return pathinfo($path, PATHINFO_EXTENSION) ?: null;
```

**Review by atomiix — DISMISSED** [review]
Just a small simplification but not mandatory ;)

**Comment by matks** [comment]
Re-running sanity test failing on PHP8.1

**Review by jolelievre — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by khouloudbelguith — APPROVED** [review]
Hi @FabienPapet,

Automated tests => OK :heavy_check_mark: 
I checked with php8.1 all types of images => OK

https://user-images.githubusercontent.com/16067358/188643916-2cc09090-bf11-4917-a8c7-474d60488858.mp4

I checked with PHP 7.2 => OK
image

Thanks!

**Comment by matks** [comment]
Thank you @FabienPapet

**Pull request merged**